src/master.rs

```diff
@@ -517,9 +517,97 @@ impl Master {
         Ok(time)
     }
 
+    pub fn foe_read(&mut self, idx: u16, name: String) -> Result<Vec<u8>> {
+        let file_name = string_to_foe_name(name);
+        // FIXME: this is the same as in the c-implementation. Should read in chunks instead of a
+        // fixed size buffer.
+        const FOE_SIZE: usize = 0x8800;
+        let mut buf: Vec<u8> = vec![0; FOE_SIZE];
+        let mut data = ec::ec_ioctl_slave_foe_t {
+            slave_position: idx,
+            offset: 0,
+            buffer_size: FOE_SIZE as u64,
+            buffer: buf.as_mut_ptr(),
+            file_name,
+            ..Default::default()
+        };
+        ioctl!(self, ec::ioctl::SLAVE_FOE_READ, &mut data)?;
+
+        assert!(data.data_size <= FOE_SIZE as u64);
+        buf.resize(data.data_size as usize, 0x0);
+        Ok(buf)
+    }
+
+    pub fn foe_write(&mut self, idx: u16, name: String, data: &mut [u8]) -> Result<()> {
+        let file_name = string_to_foe_name(name);
+
+        let data = ec::ec_ioctl_slave_foe_t {
+            slave_position: idx,
+            offset: 0,
+            buffer_size: data.len() as u64,
+            buffer: data.as_mut_ptr(),
+            file_name,
+            ..Default::default()
+        };
+        ioctl!(self, ec::ioctl::SLAVE_FOE_WRITE, &data)?;
+
+        Ok(())
+    }
+
     // XXX missing: write_idn, read_idn
 }
 
+fn string_to_foe_name(input: String) -> [std::os::raw::c_char; 32usize] {
```

Comment:
```suggestion
fn string_to_foe_name(input: String) -> [std::os::raw::c_char; 32] {
```

These are not necessary, (array) lengths ale always usize.